src/transformers/generation_utils.py

```diff
@@ -1268,6 +1300,9 @@ def greedy_search(
             if unfinished_sequences.max() == 0:
                 break
 
+            if stopping_criteria(input_ids, scores):
```

Comment:
IMO, we should add the following lines in **all** sub-generate methods for backwards compatibility:

For backwards compatibility IMO, we should add something like:

```python
if max_length is not None and # check that no max length logits processor is in StoppingCriteriaList:
   stopping_criteria.append(MaxLengthCriteria(max_length))
```

Reply:
Regarding the tests, I think we should check that passing `max_length` and no `StoppingCriteriaList` list object yields the same result as not passing `max_length`, but passing a `StoppingCriteriaList` with a `MaxLengthCriteria`

Reply:
But it's impossible, we currently still require `max_length`. So no one can call without passing `max_length`. It won't work (yet).

What I've done is adding a check that if some `stopping_criteria` along a `max_length` and `stopping_critieria.max_length` does not exist or is different from `max_length` I would :
- Warn if different
- Silently add MaxLengthCriteria if missing
- No op if both exists and are the same.

Reply:
Ok, yeah I'm fine with it!